File: ChronoSync.Android/app/src/main/java/com/chronosync/models/OffsetResult.kt
package com.chronosync.models

/**
 * Result of Theil-Sen regression: videoTime = slope * timerValue + intercept
 */
data class RegressionResult(
    val slope: Double,
    val intercept: Double
)

/**
 * Final offset calculation result for a video.
 */
data class OffsetResult(
    val videoId: String,
    val offsetSeconds: Double,
    val regression: RegressionResult?,
    val calibrationPoints: List<CalibrationPoint>,
    val hasOcr: Boolean
)

File: ChronoSync.Windows/Models/OffsetResult.cs
namespace ChronoSync.Windows.Models;

/// <summary>
/// Theil-Sen regression result: videoTime = slope * timerValue + intercept
/// </summary>
public sealed class TheilSenResult
{
    public double Slope { get; init; }
    public double Intercept { get; init; }
}

/// <summary>
/// Per-video offset calculation result.
/// </summary>
public sealed class OffsetResult
{
    public string VideoId { get; init; } = string.Empty;
    public double OffsetSeconds { get; init; }
    public int CalibrationPointCount { get; init; }
    public bool IsValid => CalibrationPointCount >= 2;
}
